doc/guides/investigating_native_memory_leak.md

```diff
@@ -0,0 +1,434 @@
+==28993== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
+==28993== Using valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
+==28993== Command: node test.js
+==28993==
+==28993== Use of uninitialised value of size 8
+==28993==    at 0x12F2279: ??? (in /home/user1/valgrind/node-v12.14.1-linux-x64/bin/node)
+==28993==    by 0x12F68A3: ??? (in /home/user1/valgrind/node-v12.14.1-linux-x64/bin/node)
+==28993==    by 0x12F68A3: ??? (in /home/user1/valgrind/node-v12.14.1-linux-x64/bin/node)
+==28993==    by 0x12F68A3: ??? (in /home/user1/valgrind/node-v12.14.1-linux-x64/bin/node)
+==28993==    by 0x12F68A3: ??? (in /home/user1/valgrind/node-v12.14.1-linux-x64/bin/node)
+==28993==    by 0x12F68A3: ??? (in /home/user1/valgrind/node-v12.14.1-linux-x64/bin/node)
+==28993==    by 0x12F68A3: ??? (in /home/user1/valgrind/node-v12.14.1-linux-x64/bin/node)
+==28993==    by 0x12F3E9C: ??? (in /home/user1/valgrind/node-v12.14.1-linux-x64/bin/node)
+==28993==    by 0x12F3C77: ??? (in /home/user1/valgrind/node-v12.14.1-linux-x64/bin/node)
+==28993==    by 0xC7C9CF: v8::internal::(anonymous namespace)::Invoke(v8::internal::Isolate*, v8::internal::(anonymous namespace)::InvokeParams const&) (in /home/user1/valgrind/node-v12.14.1-linux-x64/bin/node)
+==28993==    by 0xC7CE87: v8::internal::Execution::Call(v8::internal::Isolate*, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, int, v8::internal::Handle<v8::internal::Object>*) (in /home/user1/valgrind/node-v12.14.1-linux-x64/bin/node)
+==28993==    by 0xB4CF3A: v8::Function::Call(v8::Local<v8::Context>, v8::Local<v8::Value>, int, v8::Local<v8::Value>*) (in /home/user1/valgrind/node-v12.14.1-linux-x64/bin/node)
+==28993==
+--28993-- WARNING: unhandled amd64-linux syscall: 332
+--28993-- You may be able to write your own handler.
+--28993-- Read the file README_MISSING_SYSCALL_OR_IOCTL.
+--28993-- Nevertheless we consider this a bug.  Please report
+--28993-- it at http://valgrind.org/support/bug_reports.html.
+==28993==
+==28993== HEAP SUMMARY:
+==28993==     in use at exit: 6,140 bytes in 23 blocks
+==28993==   total heap usage: 12,888 allocs, 12,865 frees, 13,033,244 bytes allocated
+==28993==
+==28993== LEAK SUMMARY:
+==28993==    definitely lost: 0 bytes in 0 blocks
+==28993==    indirectly lost: 0 bytes in 0 blocks
+==28993==      possibly lost: 304 bytes in 1 blocks
+==28993==    still reachable: 5,836 bytes in 22 blocks
+==28993==         suppressed: 0 bytes in 0 blocks
+==28993== Rerun with --leak-check=full to see details of leaked memory
+==28993==
+==28993== For counts of detected and suppressed errors, rerun with: -v
+==28993== Use --track-origins=yes to see where uninitialised values come
+```
+
+This reports that Node.js is not _completely_ clean as there is some memory
+that was allocated but not freed when the process shut down. It is often
+impractical/not worth being completely clean in this respect and Node.js
```

Comment:
Could we add the 'why' around `It is often
impractical/not worth being completely clean in this respect` so that people not familiar with the topic will at least get a pointer of where to look? Maybe something like `most modern operating systems will clean up the memory of the process after the shutdown while doing this manually for _all_ of the cases may have a negative impact on the code complexity and shutdown times. Nevertheless, this does _not_ mean that it's fine to leave not freed memory, just that in some cases it's a necessary evil.` (my wording could obviously be better)

Reply:
@lundibundi updated